orderer/common/throttle/ratelimit.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package throttle
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// Clock observes time
+type Clock interface {
+	Now() time.Time
+	After(d time.Duration) <-chan time.Time
+}
+
+type SharedRateLimiter struct {
+	// Time is how this rate limiter observes time
+	Time Clock
+	// Limit is the total limit of throughput for all clients
+	Limit int
+	// InactivityTimeout is the inactivity time after which a client is pruned from memory
+	InactivityTimeout time.Duration
+	// state
+	once    sync.Once
+	stop    chan struct{}
+	lock    sync.RWMutex
+	clients map[string]*clientState
+}
+
+type clientState struct {
+	lock       sync.Mutex
+	lastSeen   atomic.Value
+	budget     chan struct{}
+	budgetUsed uint32
```

Comment:
maybe make the variable budgetUsed atomic.Uint32

Reply:
why? it's just a wrapper for uint32, it doesn't add much.

Reply:
So that those who will later accompany the code, could not just do simple mathematical operations (+, -, ++, --, =).

Reply:
The wrapper creates a new data type that will limit the opreations that can be produced.

But if you leave it as it is, that's fine with me.

Reply:
Thanks for your agreement